authors/apps/comments/views.py

```diff
@@ -0,0 +1,147 @@
+        try:
+            comment = Comment.objects.get(pk=self.kwargs['id'])
+            serializer = self.serializer_class(comment)
+            return Response(serializer.data,
+                            status=status.HTTP_200_OK)
+        except Comment.DoesNotExist:
+            return Response({"errors": "Specified Comment does not exist."}, status=status.HTTP_404_NOT_FOUND)
+
+    def post(self, request, *args, **kwargs):
+        """
+        Handles user replying to an existing comment.
+        :param request:
+        :param args:
+        :param kwargs:
+        :return: [reply]
+        """
+
+        try:
+            article = Articles.objects.get(slug=self.kwargs['slug'])
+            comment = request.data.get('comment', {})
+            comment['parent'] = self.kwargs['id']
+            serializer = self.serializer_class(data=comment)
+            serializer.is_valid(raise_exception=True)
+            serializer.save(author=request.user, article=article)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        except Articles.DoesNotExist:
+            return Response({"errors": "Specified Article does not exist."}, status=status.HTTP_404_NOT_FOUND)
+
+    def patch(self, request, *args, **kwargs):
+        """
+        Handles updating a comment if author is same as the requester
+        :param request:
+        :param args:
+        :param kwargs:
+        :return: [updated comment]
+        """
+        try:
+            article = Articles.objects.get(slug=self.kwargs['slug'])
+        except Articles.DoesNotExist:
+            return Response({"errors": "Specified Article does not exist."}, status=status.HTTP_404_NOT_FOUND)
+        try:
+            comment = Comment.objects.get(id=self.kwargs['id'])
+            self.check_object_permissions(request, comment)
+            serializer = self.serializer_class(comment, data=request.data.get('comment', {}), partial=True)
+            serializer.is_valid(raise_exception=True)
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        except Comment.DoesNotExist:
+            return Response({"errors": "Specified Comment does not exist."}, status=status.HTTP_404_NOT_FOUND)
+
+    def delete(self, request, **kwargs):
+        """
+        Handles method the user to delete his/her comment
+        :param request:
+        :param kwargs:
+        :return: {success: message}
+        """
+        try:
+            article = Articles.objects.get(slug=self.kwargs['slug'])
+        except Articles.DoesNotExist:
+            return Response({"errors": "Specified Article does not exist."}, status=status.HTTP_404_NOT_FOUND)
+        try:
+            comment = Comment.objects.get(pk=self.kwargs['id'])
+            self.check_object_permissions(request, comment)
+            comment.delete()
+            return Response({'message': "Deletion successful."}, status.HTTP_204_NO_CONTENT)
```

Comment:
`204` responses should not contain a message body

Reply:
> @bl4ck4ndbr0wn good job with this. Look into the issues raised, including the following:
> Improve test coverage
> `GET` - `/api/articles/<slug>/comments/`:
> 
> * Repetition of comments in response ie. A child is returned under a parent as well as on its own

In regard to test coverage currently, I have tested everything inside the comments application.
I have uploaded an image showing the coverage in my feature.
![image](https://user-images.githubusercontent.com/33542127/55937456-b329c700-5c41-11e9-9a36-b8ebedefa569.png)

Maybe to add the coverage if it is ok, i can test other apps in the project folder.